PlasmaMVP/throughputOptimization.md

```diff
@@ -0,0 +1,52 @@
+### Increase Throughput of Plasma
+
+Rather than submitting a block to the Ethereum rootchain and waiting for presumed finality before submitting the next block; we propose changes to the submitBlock function that allow blocks to be added before the previous blocks get finalized. We effectively allow the blocks to be added "asynchronously".
+
+The submitBlock function changes to:
+``submitBlock(blockNum, root, prevRoot, sig)``
+
+blockNum: The blocknumber that is being submitted
+
+root: Merkle root of transaction tree in block
+
+prevRoot: Merkle root of transaction tree in previous block (blockNum - 1)
+
+sig: Signature by the authority over the hash of blockNum, root, prevRoot.
+
+The contract keeps track of the largest blocknumber seen, call this ``latestBlock`` as well as a childchain mapping from ``(uint blocknumber => Block struct)``
+
+For a given call of submitBlock:
+
+#### Case 1: Submitting a new Block
+If the blockNum is ``1 + latestBlock``, we simply check that ``ChildChain[latestBlock].root == prevRoot``
+
+#### Case 2: Filling in reorged blocks in the middle
+If a block (with blocknumber k < latestBlock) in the middle gets reorged out of existence, one can fill in the gap by creating ``submitBlock(k, kth_block_root, k-1th_block_root, sig)``
```

Comment:
why do we need k-1th_block_root?

Reply:
Same issue as above.

Reply:
We only need k-1 assuming, k-1th block has not been reorged out?

Reply:
No. In case 2, all we are using to validate the submitBlock transaction is ``ChildChain[k + 1].prevRoot == kth_root``

However, we still need to include k-1th_block_root to fill in our Block struct. And because the k-1th block may disappear in the future, in which case we will use ChildChain[k].prevRoot to validate any transaction that tries to fill in the gap.